libs/remix-ui/workspace/src/lib/actions/index.ts

```diff
@@ -44,19 +60,67 @@ export const initWorkspace = (filePanelPlugin) => async (reducerDispatch: React.
       dispatch(setCurrentWorkspace('code-sample'))
       const filePath = await loadWorkspacePreset('code-template')
       plugin.on('editor', 'editorMounted', async () => await plugin.fileManager.openFile(filePath))
-    } else {
-      if (workspaces.length === 0) {
-        await createWorkspaceTemplate('default_workspace', 'remixDefault')
-        plugin.setWorkspace({ name: 'default_workspace', isLocalhost: false })
-        dispatch(setCurrentWorkspace('default_workspace'))
-        await loadWorkspacePreset('remixDefault')
-      } else {
-        if (workspaces.length > 0) {
-          workspaceProvider.setWorkspace(workspaces[workspaces.length - 1])
-          plugin.setWorkspace({ name: workspaces[workspaces.length - 1], isLocalhost: false })
-          dispatch(setCurrentWorkspace(workspaces[workspaces.length - 1]))
+    } else if (window.location.pathname && window.location.pathname !== '/') {
+      let route = window.location.pathname
+      if (route.startsWith('/address/0x') && route.length === 51) {
+        const contractAddress = route.split('/')[2]
+        plugin.call('notification', 'toast', `Looking for contract address ${contractAddress} on different networks`)
+        let data
+        try {
+          let etherscanKey = await plugin.call('config', 'getAppParameter', 'etherscan-access-token')
+          if (!etherscanKey) etherscanKey = '2HKUX5ZVASZIKWJM8MIQVCRUVZ6JAWT531'
+          const networks = [
+            {id: 1, name: 'mainnet'},
+            {id: 3, name: 'ropsten'},
+            {id: 4, name: 'rinkeby'},
+            {id: 42, name: 'kovan'},
+            {id: 5, name: 'goerli'}
+          ]
+          plugin.call('notification', 'toast', `Looking for contract address ${contractAddress} on different networks`)
+          let found = false
+          for (const network of networks) {
+            const target = `/${network.name}`
+            try{
+              data = await fetchContractFromEtherscan(plugin, network, contractAddress, target, etherscanKey)
+            } catch (error) {
+              if ((error.message.startsWith('contract not verified on Etherscan') || error.message.startsWith('unable to retrieve contract data')) && network.id !== 5)
+                continue
+              else {
+                if (!found) await basicWorkspaceInit(workspaces, workspaceProvider)
+                break
+              }
+            }
+            found = true
+            await createWorkspaceTemplate('etherscan-code-sample', 'code-template')
+            plugin.setWorkspace({ name: 'etherscan-code-sample', isLocalhost: false })
+            dispatch(setCurrentWorkspace('etherscan-code-sample'))
+            const filePath = Object.keys(data.compilationTargets)[0]
+            await workspaceProvider.set(filePath, data.compilationTargets[filePath]['content'])
```

Comment:
the path should also contains the contract address

Reply:
updated